src/codec.rs

```diff
@@ -307,7 +307,20 @@ pub trait Decode: Sized {
 	const TYPE_INFO: TypeInfo = TypeInfo::Unknown;
 
 	/// Attempt to deserialise the value from input.
-	fn decode<I: Input>(value: &mut I) -> Result<Self, Error>;
+	fn decode<I: Input>(input: &mut I) -> Result<Self, Error>;
+
+	/// Attempt to skip the encoded value from input.
+	fn skip<I: Input>(input: &mut I) -> Result<(), Error> {
+		Self::decode(input).map(|_| ())
+	}
+
+	/// Return the fix size of the encoded type, if any.
+	///
+	/// If it returns some size then it is the size of any encoded value of the type.
+	/// Otherwise the type encoding may or may not have fixed size.
+	fn fix_encoded_size() -> Option<usize> {
+		None
+	}
```

Comment:
```suggestion
	/// Returns `Some(_)` iff the type has a fixed encoded size.
	///
	/// This means any possible encoded value of this type has 
	/// the same size when being encoded.
	fn encoded_fixed_size() -> Option<usize> {
		None
	}
```

Reply:
`iff` means it is an equivalence https://en.wikipedia.org/wiki/If_and_only_if
But here (at least because of the default implementation) some type can have a fixed encoded size and return None.

So the doc should say: `if it returns some then it is the fixed encoded size` or maybe but I'm not sure: `Returns some only if the type has fixed encoded size`.
I'm struggling to find a good way to express it

Reply:
I gave it a try in https://github.com/paritytech/parity-scale-codec/pull/243/commits/3f3a43298cf849820dc48b14646acbb579d5c55e